frontend/src/components/common/LoginBtn/LoginModal/LoginModal.jsx

```diff
@@ -0,0 +1,231 @@
+        );
+        if (response.status === 200) {
+          handleSuccessfulLogin();
+        }
+      } catch (error) {
+        if (error.response && error.response.status === 400) {
+          setPasswordError('Невірна пошта та/або пароль!');
+        }
+      }
+    }
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setSubmitted(true);
+    setEmailError('');
+    await login();
+  };
+  const handleGoogleLogin = async (token) => {
+    await axios.post('https://api.imperiaholoda.com.ua:4446/v1/auth/google', {
+      token,
+    });
+    localStorage.setItem('authToken', token);
+    handleSuccessfulLogin();
+  };
+
+  const handleResetPassword = async () => {
+    if (validateEmail(email)) {
+      try {
+        /* eslint-disable max-len */
+        const response = await axios.post(
+          `https://api.imperiaholoda.com.ua:4446/v1/users/resetPassword/${email}`,
+        );
+
+        if (response.status === 200) {
+          setResetPasswordClicked(true);
+          toggleReset();
+        }
+      } catch (error) {
+        if (error.response && error.response.status === 404) {
+          setResetEmailError('Електрона пошта не зареєстрована!');
+        }
+      }
+    }
+  };
+  const toggleReset = () => {
+    setResetPassword((prev) => !prev);
+  };
+
+  return (
+    <>
+      {!loginSuccess && (
+        <Modals type="login" onClick={toggleLogin}>
+          <div className={styles.header}>
+            <div className={styles.headerContent}>
+              <p className={styles.loginBtn}>Вхід</p>
+              <p className={styles.registerBtn} onClick={toggleRegister}>
+                Реєстрація
+              </p>
+            </div>
+            <div className={styles.closeBtn}>
+              <IconButton icon={<ICONS.closeMobile />} onClick={toggleLogin} />
+            </div>
+          </div>
+
+          <div className={styles.formContainer}>
+            <form onSubmit={handleSubmit}>
+              <div className={styles.dataContainer}>
+                <div className={styles.emailContainer}>
+                  <label className={styles.labelData} htmlFor="email">
+                    Ел. пошта {submitted && !email && <span>*</span>}
+                  </label>
+
+                  <input
+                    id="email"
+                    className={styles.dataInput}
+                    type="email"
+                    pattern="^[a-zA-Z0-9._-]+@[a-zAZ.-]+\.[a-zA-Z]{2,4}$"
+                    placeholder="Введіть електронну пошту"
+                    value={email}
+                    onChange={(e) => {
+                      setEmail(e.target.value.trim());
+                      setEmailError('');
+                    }}
+                  />
+                  {submitted && emailError && (
+                    <p className={styles.errorMessage}>{emailError}</p>
+                  )}
+                  {submitted && !email && (
+                    <p className={styles.errorMessage}>
```

Comment:
same here. check for all error cases inside handleSubmit function and set it to emailError or resetEmailError. But keep just 1 variable, no need for both of them

Reply:
done